napari/_qt/qt_range_slider_popup.py

```diff
@@ -0,0 +1,168 @@
+from functools import partial
+
+from qtpy.QtCore import Qt, QPoint
+from qtpy.QtGui import QDoubleValidator, QFontMetrics, QFont
+from qtpy.QtWidgets import QHBoxLayout, QLineEdit
+
+from .qt_modal import QtPopup
+from .qt_range_slider import QHRangeSlider, QVRangeSlider
+from .utils import qt_signals_blocked
+
+
+class LabelEdit(QLineEdit):
+    def __init__(self, value='', parent=None, get_pos=None):
+        """Helper class to position LineEdits above the slider handle
+
+        Parameters
+        ----------
+        value : str, optional
+            starting value, by default ''
+        parent : QRangeSliderPopup, optional
+            required for proper label positioning above handle, by default None
+        get_pos : callable, optional
+            function that returns the position of the appropriate slider handle
+            by default None
+        """
+        super().__init__(value, parent=parent)
+        self.fm = QFontMetrics(QFont("", 0))
+        self.setObjectName('slice_label')
+        self.min_width = 30
+        self.max_width = 200
+        self.setCursor(Qt.IBeamCursor)
+        self.setValidator(QDoubleValidator(-9999999, 9999999, 4))
```

Comment:
I don't love these magic values....?

Reply:
silly me... can just instantiate a [QDoubleValidator](https://doc.qt.io/qt-5/qdoublevalidator.html#QDoubleValidator) without any values to achieve the desired effect.